aptos-move/aptos-vm/src/move_vm_ext/session/mod.rs

```diff
@@ -98,6 +104,9 @@ where
             resolver,
         ));
         extensions.add(RandomnessContext::new());
+        extensions.add(BulletproofContext::new(
+            !features.is_enabled(FeatureFlag::UNRESTRICTED_BULLETPROOFS_BATCH_NATIVES),
+        ));
         extensions.add(NativeTransactionContext::new(
```

Comment:
Btw, this stores `Option<UserTransactionContext>` -> `entry_function_payload: Option<EntryFunctionPayload>` -> `AccountAddress` (there is also a variant for multisig, if it is a view function or a script, stores `None`) so you can actually use this to enable/disable uses once at session creation and avoid marking when calling entry function in AptosVM.